__tests__/app.test.js

```diff
@@ -145,15 +144,37 @@ describe("/api/articles/:article_id/comments", () => {
       .get("/api/articles/lizzie/comments")
       .expect(400)
       .then(({ body }) => {
-        expect(body.msg).toBe("Sorry, try again!")
+        expect(body.msg).toBe("Sorry - not found!")
       })
     })
     test("404: throws error when article id is correct type but not on the table", () => {
       return request(app)
       .get("/api/articles/500/comments")
       .expect(404)
       .then(({ body }) => {
-        expect(body.err.msg).toBe("Sorry, I couldn't find that!")
+        expect(body.msg).toBe("Sorry, I couldn't find that article!")
       })
     })
-  })
\ No newline at end of file
+  })
+  describe("/api/articles/:article_id/comments POST", () => {
+    test("201: successfully inserts a comment", () => {
+      return request(app)
+      .post("/api/articles/2/comments")
+      .expect(201)
+      .send({username: "lurker", body: "I just got engaged!!!"})
+      .then(({ body }) => {
+        expect(body.msg).toBe("comment posted!")
+        expect(body.comment).toBe("I just got engaged!!!")
+      })
+      })
+      test("400: throws error when given a bad article id", () => {
+        return request(app)
+        .get("/api/articles/lizzie/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Sorry - not found!")
+        })
+      })
+    })
```

Comment:
Nice start to error handling as well, but the message here is not very accurate. This is essentially a bad request so the message should instead indicate this instead of not found as we cannot even start looking since the article_id is not a number.

There are also more things that can go wrong so we need to test for them as well:

When given an article_id that is valid but does not exist what should we expect? E.g /api/articles/999/comments

When we are not sending the body containing the message/username what should we expect? E.g. {username: "lurker"}

When given a username that does not exist what should we expect? E.g. {username: "bob", body: "I am bob"}